### PR TITLE
IFT E-field reconstruction

### PR DESCRIPTION
**Pull request by christophwelling**
Adds a module to perform electric field reconstructions using IFT.

### CODE REVIEW

File: NuRadioReco/modules/neutrinoVertexReconstructor/neutrino2DVertexReconstructor.py [hunk 1]
```diff
@@ -35,16 +39,30 @@ def __init__(self, lookup_table_location):
...
+            ['direct', 'reflected'],
+            ['reflected', 'direct'],
+            ['direct', 'refracted'],
+            ['refracted', 'direct'],
+            ['reflected', 'refracted'],
+            ['refracted', 'reflected']
+        ]
 
-    def begin(self, station_id, channel_ids, detector):
+    def begin(self, station_id, channel_ids, detector, passband=None, template=None, output_path=None):
```
Comment (cg-laser) [inline NuRadioReco/modules/neutrinoVertexReconstructor/neutrino2DVertexReconstructor.py:65]: template and output_path not documented

File: NuRadioReco/modules/neutrinoVertexReconstructor/neutrino2DVertexReconstructor.py [hunk 2]
```diff
@@ -73,14 +94,17 @@ def begin(self, station_id, channel_ids, detector):
...
             channel_z = abs(detector.get_relative_position(station_id, channel_id)[2])
             if channel_z not in self.__lookup_table.keys():
                 f = NuRadioReco.utilities.io_utilities.read_pickle('{}/lookup_table_{}.p'.format(self.__lookup_table_location, int(abs(channel_z))))
                 self.__header[int(channel_z)] = f['header']
                 self.__lookup_table[int(abs(channel_z))] = f['antenna_{}'.format(channel_z)]
+        self.__template = template
+        self.__output_path = output_path
 
-    def run(self, station, max_distance, z_width, grid_spacing, direction_guess=None, debug=False):
+    def run(self, event, station, max_distance, z_width, grid_spacing, direction_guess=None, debug=False, use_dnr=False):
```
Comment (cg-laser) [inline NuRadioReco/modules/neutrinoVertexReconstructor/neutrino2DVertexReconstructor.py:107]: use_dnr not documented

**Review by cg-laser — COMMENTED** [review]
I didn't check the code thoroughly, I think it is also too much code to do so... but as this is an independent reco module, it is not that critical. 
I added a few inline comments about missing documentation. 

it would be nice to have a working example with this module so that people who want to start using the module know what to do. Especially what modules are expected to come before this reco module. Ideally something that we can also integrate into a unit test.

**Comment by christophwelling** [comment]
I added the parameters to the documentation.
I also added an example to run the IFT reconstruction on a cosmic ray event.

**Review by cg-laser — APPROVED** [review]

**Pull request merged**